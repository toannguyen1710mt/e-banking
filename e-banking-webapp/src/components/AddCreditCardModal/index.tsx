'use client';

import { useEffect, useState, useTransition } from 'react';
import Image from 'next/image';
import { z } from 'zod';
import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';
import { AuthError, Session } from 'next-auth';

// Constants
import { ERROR_MESSAGES, IMAGES } from '@/constants';

// Interfaces
import { IAccount } from '@/interfaces';

// Actions
import { addNewCardByAccountId } from '@/actions';

// Schemas
import { CreditCardSchema } from '@/schemas';

// Services
import { getAccountsByUserId } from '@/services';

// Components
import * as WizardForm from '@/components/common/WizardForm';
import { Modal } from '../common';
import { AddCreditCard } from './AddCreditCard';
import { ConfirmAddCard } from './ConfirmAddCard';
import { toastStore } from '@/utils';

interface IAddCreditCardModalProps {
  session: Session;
  isOpen: boolean;
  onClose: () => void;
}

type FormValues = z.infer<typeof CreditCardSchema>;

export const AddCreditCardModal = ({
  isOpen,
  session,
  onClose,
}: IAddCreditCardModalProps) => {
  const [accounts, setAccounts] = useState<IAccount[]>([]);
  const [isPending, startTransition] = useTransition();

  const form = useForm<FormValues>({
    resolver: zodResolver(CreditCardSchema),
    defaultValues: {
      cardInfo: {
        holderName: '',
        cardNumber: '',
        expireAt: '',
        ccv: '',
      },
      walletType: 'Main',
    },
    reValidateMode: 'onBlur',
    mode: 'onBlur',
  });

  useEffect(() => {
    const fetchData = async () => {
      try {
        const result = await getAccountsByUserId(session.user.id);

        setAccounts(result || []);
      } catch (error) {
        if (error instanceof AuthError) {
          throw ERROR_MESSAGES.GET_ERROR;
        }
      }
    };

    fetchData();
  }, [session.user.id]);

  const onSubmit = async (data: FormValues) => {
    const { holderName, cardNumber, expireAt, ccv } = data.cardInfo;

    const selectedAccount = accounts.find(
      (account) => account.type === data?.walletType,
    );

    try {
      await addNewCardByAccountId(selectedAccount?.documentId as string, {
        holderName,
        cardNumber,
        expireAt,
        ccv,
      });

      startTransition(() => {
        toastStore.showToast(
          ERROR_MESSAGES.ADD_CARD_SUCCESS,
          'success',
          'top-center',
        );

        return onClose();
      });
    } catch (error) {
      toastStore.showToast(
        ERROR_MESSAGES.ADD_CARD_FAILED,
        'error',
        'top-center',
      );

      if (error instanceof AuthError) {
        throw ERROR_MESSAGES.ADD_CARD_FAILED;
      }
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      onClose={onClose}
      size='4xl'
      placement='center'
      classNames={{
        base: 'p-0 rounded-xl bg-primary-200',
      }}
    >
      <div className='flex items-center justify-between'>
        <Image
          className='max-w-[50%] flex-1 px-[31px] py-[87px]'
          src={IMAGES.ADD_CREDIT_CARD_IMAGE}
          width={333}
          height={280}
          alt='Add Credit Card'
        />

        <div className='min-h-[500px] flex-1 rounded-xl bg-white px-4 py-8'>
          <WizardForm.Root
            schema={CreditCardSchema}
            form={form}
            className='flex grow flex-col'
            onSubmit={onSubmit}
          >
            <WizardForm.Step name='cardInfo' key='cardInfo'>
              <AddCreditCard />
            </WizardForm.Step>

            <WizardForm.Step
              name='confirmationDetails'
              key='confirmationDetails'
            >
              <ConfirmAddCard accounts={accounts} isPending={isPending} />
            </WizardForm.Step>
          </WizardForm.Root>
        </div>
      </div>
    </Modal>
  );
};
