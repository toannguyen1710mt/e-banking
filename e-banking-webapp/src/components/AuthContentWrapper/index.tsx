'use client';

import { ReactNode } from 'react';
import { usePathname } from 'next/navigation';
import Link from 'next/link';
import Image from 'next/image';

// Constants
import { IMAGES, ROUTES } from '@/constants';

// Interfaces
import { TEXT_SIZE, TEXT_VARIANT } from '@/interfaces';

// Components
import { Text } from '../common';

interface AuthContentWrapperProps {
  textFooter?: string;
  textHeading: ReactNode;
  formContent: ReactNode;
  subheading?: string;
}

export const AuthContentWrapper = ({
  textHeading,
  formContent,
  textFooter,
  subheading,
}: AuthContentWrapperProps) => {
  const pathname = usePathname();

  const isOnSignIn = pathname === ROUTES.SIGN_IN;

  const linkDetails = {
    text: isOnSignIn ? 'Register' : 'Sign In',
    href: isOnSignIn ? ROUTES.SIGN_UP : ROUTES.SIGN_IN,
  };

  const classLogo = isOnSignIn
    ? 'mt-0 md:mt-[108px] pb-[56px] md:pb-9'
    : 'pb-[14px] lg:pb-9';
  const classTextHeading = isOnSignIn ? 'mb-7' : 'mb-0 md:mb-7';

  return (
    <div className='mx-auto flex flex-1 flex-col'>
      {/* Logo */}
      <Image
        className={`${classLogo}`}
        src={IMAGES.LOGO}
        width={104}
        height={36}
        alt='Logo EBanking'
      />

      <div className='max-w-[406px] text-center md:text-left'>
        <Text
          as='h1'
          size={TEXT_SIZE['4XL']}
          variant={TEXT_VARIANT.DEFAULT}
          className={`${classTextHeading}`}
        >
          {textHeading}
        </Text>
        <Text
          size={TEXT_SIZE.BASE}
          variant={TEXT_VARIANT.INFO}
          className='mb-4 font-normal leading-[22px]'
        >
          {subheading}
        </Text>
      </div>

      {/* Form */}
      <div className='mx-auto mb-[17px] max-w-[325px] md:mx-0'>
        {formContent}
      </div>

      {textFooter && (
        <Text
          size={TEXT_SIZE.SM}
          variant={TEXT_VARIANT.DEFAULT}
          className='text-center leading-[20px] md:text-left'
        >
          {textFooter}&nbsp;
          <Link
            href={linkDetails.href}
            className='leading-[20px] text-primary-200'
          >
            {linkDetails.text}
          </Link>
        </Text>
      )}
    </div>
  );
};
