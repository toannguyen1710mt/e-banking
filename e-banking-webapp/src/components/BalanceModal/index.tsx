'use client';

import { useCallback, useEffect, useMemo, useState } from 'react';
import { CircularProgress } from '@nextui-org/react';
import { Session } from 'next-auth';

// Constants
import { createExpenseAnalysisOptions } from '@/constants';

// Interfaces
import { IAccount, ICard, TEXT_SIZE, TEXT_VARIANT } from '@/interfaces';

// Utils
import { formatNumberWithCommas } from '@/utils';

// Services
import { getAccountsByUserId, getListCardByAccountId } from '@/services';

// Mocks
import { MOCK_SERIES_EXPENSE_ANALYSIS } from '@/mocks';

// Components
import { Modal, Text } from '../common';
import { MyCards } from '../MyCards';
import { ExpenseAnalysis } from '../ExpenseAnalysis';

interface BalanceModalProps {
  session: Session;
  isOpen?: boolean;
  totalInvestment?: string;
  totalIncome?: string;
  onClose: () => void;
}

const BalanceModal = ({
  session,
  totalInvestment = '170,000',
  totalIncome = '670,000',
  isOpen = false,
  onClose,
}: BalanceModalProps) => {
  const [accounts, setAccounts] = useState<IAccount[]>([]);
  const [cardData, setCardData] = useState<ICard>({} as ICard);
  const [selectedAccount, setSelectedAccount] = useState<IAccount>(
    {} as IAccount,
  );

  useEffect(() => {
    const fetchData = async () => {
      try {
        const result = await getAccountsByUserId(session.user.id);

        if (result?.length) {
          setAccounts(result);
          setSelectedAccount(result[0]);
          const documentId = result[0]?.documentId;

          if (documentId) {
            const { cards } = await getListCardByAccountId(
              documentId as string,
            );

            setCardData(cards[0] || {});
          }
        }
      } catch (error) {
        console.error('Error fetching data:', error);
      }
    };

    fetchData();
  }, [session?.user?.accounts, session?.user?.id]);

  const handleCardSelect = useCallback((account: IAccount) => {
    setSelectedAccount(account);
  }, []);

  const totalBalance = useMemo(
    () => accounts?.reduce((total, account) => total + account.balance, 0),
    [accounts],
  );

  return (
    <Modal
      isOpen={isOpen}
      onClose={onClose}
      placement='center'
      classNames={{
        base: 'max-w-xs lg:max-w-4xl p-0 rounded-xl',
        closeButton: 'z-50',
      }}
    >
      <div className='absolute left-4 top-4'>
        <Text as='h4' className='text-base font-medium text-primary-200'>
          Good Evening,{' '}
          <Text
            as='span'
            variant={TEXT_VARIANT.DEFAULT}
            className='font-normal'
          >
            {session?.user?.username || ''}
          </Text>
        </Text>
      </div>
      <div className='flex flex-col rounded-xl bg-lightGraySolid lg:flex-row'>
        <div className='flex flex-1 flex-col rounded-xl bg-white pb-[48px] pl-3 pr-[23px] pt-[56px]'>
          <div className='flex flex-col items-start justify-between gap-2 md:flex-row'>
            <div className='flex flex-col gap-2'>
              <Text
                as='h5'
                className='text-sm font-semibold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                Current Balance
              </Text>
              <Text
                as='span'
                className='font-extrabold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                ${formatNumberWithCommas(selectedAccount?.balance || 0)}
                <Text
                  as='span'
                  size={TEXT_SIZE['XS']}
                  variant={TEXT_VARIANT.DEFAULT}
                  className='font-semibold opacity-50'
                >
                  /${totalIncome}
                </Text>
              </Text>
            </div>
            <div className='flex flex-col gap-2'>
              <Text
                as='h5'
                className='text-sm font-semibold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                Total Investments
              </Text>
              <Text
                as='span'
                className='font-extrabold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                ${totalInvestment}
              </Text>
            </div>
          </div>

          <div className='relative mx-auto mt-9 flex h-[100px] w-[100px] items-center justify-center'>
            <CircularProgress
              classNames={{
                svg: 'w-full h-full drop-shadow-md',
                indicator: 'stroke-[#2A9D8F]',
                track: 'stroke-[#2A9D8F]/10',
                value: 'text-medium font-extrabold text-black',
              }}
              showValueLabel={false}
              strokeWidth={3}
              value={70}
            />
            <div className='absolute flex flex-col items-center'>
              <Text
                className='text-center text-sm font-medium'
                variant={TEXT_VARIANT.DEFAULT}
              >
                Spend
              </Text>
              <Text
                as='span'
                className='text-center font-extrabold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                70%
              </Text>
            </div>
          </div>
        </div>

        {/* My Cards and Expense Analysis */}
        <div className='flex flex-1 flex-col py-8 pl-4 pr-3'>
          <Text as='h4' variant={TEXT_VARIANT.DEFAULT}>
            My Cards
          </Text>
          <div className='my-4'>
            <MyCards
              expireDate={cardData?.expireAt}
              accounts={accounts as IAccount[]}
              onCardSelect={handleCardSelect}
            />
          </div>
          <Text
            size={TEXT_SIZE['XS']}
            variant={TEXT_VARIANT.DEFAULT}
            className='max-w-[180px] font-normal'
          >
            You have a total of{' '}
            <Text
              as='span'
              size={TEXT_SIZE['XS']}
              variant={TEXT_VARIANT.DEFAULT}
              className='font-bold'
            >
              ${formatNumberWithCommas(totalBalance || 0)}
            </Text>{' '}
            available in your wallet
          </Text>

          <div className='mt-2.5'>
            <ExpenseAnalysis
              options={createExpenseAnalysisOptions(
                formatNumberWithCommas(totalBalance || 0),
              )}
              series={MOCK_SERIES_EXPENSE_ANALYSIS}
            />
          </div>
        </div>
      </div>
    </Modal>
  );
};

export default BalanceModal;
