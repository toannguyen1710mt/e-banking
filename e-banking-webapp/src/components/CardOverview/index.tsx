'use client';

// Libs
import { useEffect, useState } from 'react';
import { Card, CardBody, useDisclosure } from '@nextui-org/react';
import { AuthError, Session } from 'next-auth';

// Constants
import { createExpenseAnalysisOptions, ERROR_MESSAGES } from '@/constants';

// Interface
import { ICard } from '@/interfaces';

// Services
import { getMainCardByUserId } from '@/services';

// Mocks
import { MASTERCARD_CHART_MOCK } from '@/mocks';

// Component
import {
  Button,
  Text,
  CreditCard,
  MasterCard,
  AddCreditCardModal,
} from '@/components';

interface ICardOverviewProps {
  session: Session;
}

export const CardOverview = ({ session }: ICardOverviewProps) => {
  const [card, setCard] = useState<ICard>({} as ICard);
  const {
    isOpen: isOpenAddCardModal,
    onOpen: onOpenAddCardModal,
    onClose: onCloseAddCardModal,
  } = useDisclosure();

  useEffect(() => {
    const fetchData = async () => {
      try {
        const card = await getMainCardByUserId(Number(session?.user.id));

        setCard(card);
      } catch (error) {
        if (error instanceof AuthError) {
          throw ERROR_MESSAGES.GET_ERROR;
        }
      }
    };

    fetchData();
  }, [session?.user?.id]);

  const { cardNumber, expireAt, holderName } = card;

  return (
    <>
      <Card className='col-span-3'>
        <CardBody className='flex flex-col justify-between gap-6 p-0 sm:flex-row'>
          <div className='flex w-full max-w-full flex-col gap-4 px-4 py-2 md:max-w-[325px]'>
            <div className='flex items-center justify-between'>
              <Text as='span' className='text-sm font-semibold text-navyBlue'>
                My Cards
              </Text>
              <Button
                variant='outline'
                color='outline'
                radius='xs'
                className='h-auto w-auto border-navyBlue px-2 py-1 text-[10px] font-medium text-navyBlue'
                onClick={onOpenAddCardModal}
              >
                Add Card
              </Button>
            </div>
            <CreditCard
              cardNumber={cardNumber}
              expireDate={expireAt}
              holderName={holderName}
              bankName='Universal Bank'
            />
          </div>
          <MasterCard
            series={MASTERCARD_CHART_MOCK}
            totalBalance={createExpenseAnalysisOptions('$540,000')}
          />
        </CardBody>
      </Card>

      {isOpenAddCardModal && (
        <AddCreditCardModal
          session={session}
          isOpen={isOpenAddCardModal}
          onClose={onCloseAddCardModal}
        />
      )}
    </>
  );
};
