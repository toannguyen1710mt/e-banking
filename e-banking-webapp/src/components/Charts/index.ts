export * from './SpendingStatistics';
export * from './BalanceStatistics';
