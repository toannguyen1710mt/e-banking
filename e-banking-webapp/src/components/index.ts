export * from './common';
export * from './icons';
export * from './Error';
export * from './NotFound';
export * from './Slideshow';
export * from './AuthContentWrapper';
export * from './LoginForm';
export * from './TargetTile';
export * from './QuickAction';
export * from './SignUpForm';
export * from './Charts';
export * from './Sidebar';
export * from './CreditCard';
export * from './Transfers';
export * from './ExpenseAnalysis';
export * from './StepProgress';
export * from './MyCards';
export * from './BalanceModal';
export * from './MyTarget';
export * from './SettingContainer/Account/ConnectedAccountsTab';
export * from './SettingContainer/General';
export * from './SettingContainer/Account/DeleteAccountTab';
export * from './ServiceCard';
export * from './SettingContainer/Account';
export * from './DueTile';
export * from './MasterCard';
export * from './CardOverview';
export * from './MetricsCard';
export * from './Skeletons';
export * from './TasksWithCalendar';
export * from './ContainerAnalytics';
export * from './AddCreditCardModal';
export * from './InformationCard';
export * from './AddCreditCardModal/AddCreditCard';
export * from './AnalyticsCard';
export * from './DailyLimit';
export * from './TransactionTable';
export * from './TransactionHistory';
export * from './ContainerTransactions';
export * from './MonthYearPicker';
