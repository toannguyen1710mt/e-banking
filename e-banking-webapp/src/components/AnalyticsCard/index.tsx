'use client';

// Libs
import { Card, CardBody, CardFooter, CardHeader } from '@nextui-org/react';

// Utils
import { formatNumberWithCommas } from '@/utils';

// Components
import { ArrowDownIcon, ArrowUpIcon, Chip, Text } from '@/components';

export interface IAnalyticsCardItem {
  title: string;
  subtitle: string;
  amount: number;
  percentageChange: number;
  isPositive?: boolean;
}

export const AnalyticsCard = ({
  title,
  subtitle,
  isPositive,
  amount,
  percentageChange,
}: IAnalyticsCardItem) => {
  const ArrowComponent = isPositive ? <ArrowUpIcon /> : <ArrowDownIcon />;

  return (
    <Card key={title} className='w-full gap-4 rounded-md p-0 shadow-none'>
      <CardHeader className='p-4 text-xs font-medium'>
        <Text>{title}</Text>
      </CardHeader>
      <CardBody className='flex flex-row items-center gap-5 overflow-y-hidden p-4'>
        <Text as='span' className='text-[2em] font-extrabold'>
          ${formatNumberWithCommas(amount)}
        </Text>
        <Chip
          startContent={ArrowComponent}
          color={isPositive ? 'success' : 'danger'}
          classNames={{
            base: 'h-5',
            content: 'font-extrabold text-4xs p-0',
          }}
        >
          {percentageChange}%
        </Chip>
      </CardBody>
      <CardFooter className='rounded-none p-4'>
        <Text className='text-2xs font-light text-transparentBlack'>
          {subtitle}
        </Text>
      </CardFooter>
    </Card>
  );
};
