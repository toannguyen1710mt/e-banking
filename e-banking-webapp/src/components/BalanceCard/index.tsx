import { Card } from '@nextui-org/react';

// Utils
import { formatNumberWithCommas } from '@/utils';

// Interfaces
import { TEXT_SIZE, TEXT_VARIANT } from '@/interfaces';

// Components
import { Text, ArrowUpRightIcon, PaymentIcon } from '@/components';

interface BalanceCardProps {
  amount: number;
  title: string;
}

export const BalanceCard = ({ amount, title }: BalanceCardProps) => (
  <Card className='h-[155px] w-full pl-[14px] pr-4 pt-[13px]'>
    <div className='mb-10 flex justify-between text-primary-200'>
      <PaymentIcon />
      <ArrowUpRightIcon />
    </div>

    <div className='flex flex-col gap-3'>
      <Text className='text-2xl font-extrabold text-primary-200' as='span'>
        ${formatNumberWithCommas(amount)}
      </Text>

      <Text size={TEXT_SIZE['2XS']} variant={TEXT_VARIANT.INFO}>
        {title}
      </Text>
    </div>
  </Card>
);
