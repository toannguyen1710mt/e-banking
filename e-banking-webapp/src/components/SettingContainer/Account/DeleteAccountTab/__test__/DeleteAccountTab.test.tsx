// Libs
import { render } from '@testing-library/react';

// Components
import { DeleteAccountTab } from '@/components';

describe('DeleteAccountTab component', () => {
  it('should match snapshot for DeleteAccountTab', () => {
    const { container } = render(<DeleteAccountTab />);

    expect(container).toMatchSnapshot();
  });
});
