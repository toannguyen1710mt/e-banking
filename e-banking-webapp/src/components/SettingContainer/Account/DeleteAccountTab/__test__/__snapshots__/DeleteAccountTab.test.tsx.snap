// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAccountTab component should match snapshot for DeleteAccountTab 1`] = `
<div>
  <div
    class="flex flex-col gap-[18px]"
  >
    <span
      class="text-primary-200 text-sm font-semibold text-navyBlue"
    >
      Delete Account
    </span>
    <div
      class="mb-8 flex gap-4"
    >
      <span
        class="text-primary-200 whitespace-nowrap text-xs font-semibold text-red"
      >
        Warning :
      </span>
      <span
        class="text-foreground-100 text-xs font-normal"
      >
        Deleting Your Account is irreversible , it deletes all your data and information
      </span>
    </div>
  </div>
  <div
    class="flex max-w-[680px] flex-col gap-[27px]"
  >
    <div
      class="group flex flex-col data-[hidden=true]:hidden w-full opacity-disabled pointer-events-none relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
      data-disabled="true"
      data-filled="true"
      data-filled-within="true"
      data-has-elements="true"
      data-has-label="true"
      data-slot="base"
    >
      <div
        class="h-full flex flex-col"
        data-slot="main-wrapper"
      >
        <div
          class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-10 min-h-10 pointer-events-none transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background !bg-inherit border border-primary-200 group-data-[focus=true]:border-primary-200 data-[hover=true]:border-secondary-200 data-[hover=true]:!bg-inherit group-data-[invalid=true]:!border-warning rounded-md"
          data-slot="input-wrapper"
          style="cursor: text;"
        >
          <label
            class="absolute origin-top-left rtl:origin-top-right subpixel-antialiased block text-foreground-500 pointer-events-none will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-foreground group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 top-1/2 -translate-y-1/2 group-data-[filled-within=true]:start-0 start-3 end-auto group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[invalid=true]:!text-primary-200 font-normal text-xs !text-black opacity-100"
            data-slot="label"
            for="react-aria-:r0:"
            id="react-aria-:r1:"
          >
            Current Password
          </label>
          <div
            class="inline-flex w-full items-center h-full box-border"
            data-slot="inner-wrapper"
          >
            <input
              aria-label="Current Password"
              aria-labelledby="react-aria-:r0: react-aria-:r1:"
              class="font-normal bg-transparent !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text group-data-[has-value=true]:text-default-foreground !text-foreground-100 placeholder:text-foreground-100 w-full h-10 py-2.5 m-0 text-sm"
              data-slot="input"
              disabled=""
              id="react-aria-:r0:"
              placeholder=" "
              type="Password"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 gap-2 opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover border-transparent bg-warning text-foreground-200 rounded w-[168px] h-10 py-3 text-xs font-semibold"
      data-disabled="true"
      disabled=""
      type="button"
    >
      Delete Account
    </button>
  </div>
</div>
`;
